crates/ide_completion/src/context.rs

```diff
@@ -487,6 +489,8 @@ impl<'a> CompletionContext<'a> {
             let for_is_prev2 = for_is_prev2(syntax_element.clone());
             (fn_is_prev && !inside_impl_trait_block) || for_is_prev2
         };
+        // todo(jake): is this the right place to put this?
+        self.param_def = find_node_at_offset(&file_with_fake_ident, offset);
```

Comment:
I think we can just replace the `is_param` field with the `param_def` and check for `is_some` where the former was used.

Assigning `param_def` should be done where `is_param` is assigned then.